ksqldb-common/src/main/java/io/confluent/ksql/util/ConsistencyOffsetVector.java

```diff
@@ -27,17 +34,73 @@
 import java.util.concurrent.locks.ReadWriteLock;
 import java.util.concurrent.locks.ReentrantReadWriteLock;
 
-public final class OffsetVectorImpl implements OffsetVector, Serializable {
+/**
+ * Represents a consistency token that is communicated between the client and server.
+ * This token is used to ensure monotonic reads for table pull queries as in: once a client has
+ * received a row with an offset x, subsequent reads will only return rows of offsets > x.
+ */
+//@JsonInclude(JsonInclude.Include.NON_ABSENT)
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class ConsistencyOffsetVector implements OffsetVector {
+
+  private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  static {
+    OBJECT_MAPPER.registerModule(new Jdk8Module());
+  }
+
+  // This field should be the first
   private int version;
+  // Topic -> Partition -> Offset
   private Map<String, Map<Integer, Long>> offsetVector;
   private ReadWriteLock rwLock;
 
-  public OffsetVectorImpl() {
+  public ConsistencyOffsetVector() {
     this.version = 0;
     this.offsetVector = new HashMap<>();
     this.rwLock = new ReentrantReadWriteLock();
   }
 
+  @SuppressFBWarnings(value = "EI_EXPOSE_REP2")
+  @JsonCreator
+  public ConsistencyOffsetVector(
+      final @JsonProperty(value = "version", required = true) int version,
+      final @JsonProperty(value = "offsetVector", required = true)
+          Map<String, Map<Integer, Long>> offsetVector
+  ) {
+    this.version = version;
+    this.offsetVector = offsetVector;
+    this.rwLock = new ReentrantReadWriteLock();
+  }
+
+  public int getVersion() {
+    return version;
+  }
+
+  @VisibleForTesting
+  public ImmutableMap<String, Map<Integer, Long>> getOffsetVector() {
+    return ImmutableMap.copyOf(offsetVector);
+  }
```

Comment:
Eventually I would think we don't want to expose all the internal of the OffsetVector, otherwise there would be no need to create at type.

Reply:
This needs to be public for Json serialization/deserialization to work